pkg/reconciler/kubernetes/tektonpipeline/transform.go

```diff
@@ -141,19 +142,14 @@ func updatePerformanceFlagsInDeployment(pipelineCR *v1alpha1.TektonPipeline) mf.
 				argStringValue := fmt.Sprintf("%v", flags[flagKey])
 				argUpdated := false
 				for argIndex, existingArg := range container.Args {
-					if existingArg == expectedArg {
-						argValueIndex := argIndex + 1
-						if len(container.Args) > argValueIndex {
-							container.Args[argValueIndex] = argStringValue
-						} else {
-							container.Args = append(container.Args, argStringValue)
-						}
+					if strings.HasPrefix(existingArg, expectedArg) {
+						container.Args[argIndex] = fmt.Sprintf("%s=%s", expectedArg, argStringValue)
```

Comment:
in the future, if these flags introduced in the pipelines manifest as array form, we have to remove those duplicates.